### PR TITLE
Deprecate old circuit drawing infrastructure

### PR DESCRIPTION
**Pull request by albi3ro**
This PR adds deprecation warnings to:

* `qml.tape.QuantumTape.draw`: Should instead use `qml.drawer.tape_text(tape)`
* `CircuitGraph.draw`: `tape_text` will not work with `CircuitGraph` objects because they do not have a `measurements` attribute.  I think this method is just developer facing anyway, I do not believe this will be a problem.
* `qml.drawer.Charsets`: We assume everybody has access to unicode now.  That is a fairly secure assumption. Even if they don't, the circuit drawer should still be readable.
* `qml.drawer.CircuitDrawer`: Replaced by `qml.drawer.tape_text` function.
* `qml.drawer.Grid`: Now we just use ordinary lists and sets to get the job done.  Enhances readability and reduces complexity.
* `qml.drawer.RepresentationResolver`: Now `Operator.label` gets the same job done.

### CODE REVIEW

File: pennylane/circuit_graph.py [hunk 1]
```diff
@@ -636,6 +637,9 @@ def draw(self, charset="unicode", wire_order=None, show_all_wires=False, max_len
         Returns:
             str: The circuit diagram representation of the ``CircuitGraph``
         """
+        warnings.warn(
+            "CircuitGraph.draw is deprecated. Please see qml.drawer.tape_text instead.", UserWarning
+        )
```
Comment (josh146) [inline pennylane/circuit_graph.py:642]: 💪 Don't forget to add a warning to the docstring as well!
E.g.,
```rest
.. warning::

    ``CircuitGraph.draw`` is deprecated. Please use :func:`~.drawer.tape_text` instead.
```

File: pennylane/drawer/circuit_drawer.py [hunk 2]
```diff
@@ -62,6 +63,9 @@ def __init__(
         max_length=None,
         _label_offsets=None,
     ):
+        warnings.warn(
+            "CircuitDrawer is deprecated. Please see qml.drawer.tape_text instead.", UserWarning
```
Comment (josh146) [inline pennylane/drawer/circuit_drawer.py:67]: @albi3ro quick question: what do you think about overloading `qml.draw()` so that it accepts tapes and qnodes?

E.g.,

```python
>>> qml.draw(tape)
>>> qml.draw(qnode)(*args, **kwargs)
```

Comment (josh146) [reply pennylane/drawer/circuit_drawer.py:67]: This means less overhead for developers/users needing to find the correct function, at the expense of a more complicated docstring.

Comment (albi3ro) [reply pennylane/drawer/circuit_drawer.py:67]: looking into it, see #2281

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -297,6 +297,10 @@
 
 <h3>Improvements</h3>
 
+* Most compilation transforms, and relevant subroutines, have been updated to
+  support just-in-time compilation with `jax.jit`.
+  [(#1894)](https://github.com/PennyLaneAI/pennylane/pull/1894/)
+
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:303]:
```suggestion
```

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -538,6 +553,6 @@ The Operator class has undergone a major refactor with the following changes:
 
 This release contains contributions from (in alphabetical order):
 
-Sam Banning, Thomas Bromley, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
+Sam Banning, Thomas Bromley, Olivia Di Matteo, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:556]:
```suggestion
Sam Banning, Thomas Bromley, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
```

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -337,6 +333,17 @@
 
 <h3>Breaking changes</h3>
 
+* Old circuit drawing infrastructure is being deprecated. This includes the following methods and classes:
```
Comment (antalszava) [inline doc/releases/changelog-dev.md:336]: This entry may fit the `Deprecations` section better, especially as these are all developer-facing functions/methods.

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -542,6 +549,6 @@ The Operator class has undergone a major refactor with the following changes:
 
 This release contains contributions from (in alphabetical order):
 
-Sam Banning, Thomas Bromley, Olivia Di Matteo, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
+Sam Banning, Thomas Bromley, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
```
Comment (antalszava) [inline doc/releases/changelog-dev.md:552]:
```suggestion
Sam Banning, Thomas Bromley, Olivia Di Matteo, Anthony Hayes, David Ittah, Josh Izaac, Christina Lee, Angus Lowe,
```

File: pennylane/transforms/decompositions/single_qubit_unitary.py [hunk 7]
```diff
@@ -14,13 +14,13 @@
...
 """
-
+import numpy as np
 import pennylane as qml
 from pennylane import math
 
 
 def _convert_to_su2(U):
-    r"""Convert a 2x2 unitary matrix to :math:`SU(2)`.
+    r"""Check unitarity of a matrix and convert it to :math:`SU(2)` if possible.
```
Comment (antalszava) [inline pennylane/transforms/decompositions/single_qubit_unitary.py:23]: It is odd, this seems to be reverting the changes from https://github.com/PennyLaneAI/pennylane/pull/1894, although that commit is already in v0.22.0-rc0 and this PR. :thinking:

**Review by antalszava — COMMENTED** [review]
Looks good so far! :+1: Two things: there's something odd with the diff with https://github.com/PennyLaneAI/pennylane/pull/1894/ and the changelog entry would better go in the Deprecations section.

**Review by antalszava — APPROVED** [review]
:100:

**Comment by albi3ro** [comment]
Given I cannot figure out why the tests on this PR run forever, and the effort to debug why the tests are hanging will be a large amount of effort for something that wasn't strictly necessary, I'm just going to close this PR without merging.
